text/trace/0157-messaging-semantic-conventions.md

```diff
@@ -0,0 +1,210 @@
+
+## Terminology
+
+To leverage existing standards, the terminology used in this document is based
+on the [CloudEvents specification](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md).
+CloudEvents is hosted by the CNCF and provides a specification for describing
+event data in common formats to provide interoperability across services,
+platforms and systems.
+
+### Message
+
+A "message" is a transport wrapper for the transfer of information. The
+information is a combination of data and metadata.  Messages may be transferred
+directly between parties or via one or more intermediaries.  A message may
+carry annotations that are directed at intermediaries on the message path.
+Messages are uniquely identifiable.
+
+In the strict sense, a _message_ is an item of data that is sent to a specific
+destination, whereas an _event_ is a signal emitted by a component upon
+reaching a given state. This document is agnostic of those differences and uses
+the term "message" in a wider sense to cover both concepts.
+
+### Producer
+
+The "producer" is a specific instance, process or device that creates and
+publishes a message. "Publishing" is the process of sending a message or batch
+to the intermediary or consumer.
+
+### Consumer
+
+A "consumer" receives the message and acts upon it. It uses the context and
+data to execute some logic, which might lead to the occurrence of new events.
+
+The consumer receives, processes, and settles a message. "Receiving" is the
+process of obtaining a message from the intermediary, "processing" is the
+process of acting on the information a message contains, "settling" is the
+process of notifying an intermediary that a message was processed successfully.
+
+### Intermediary
+
+An "intermediary" receives a message to forward it to the next receiver, which
+might be another intermediary or a consumer.
+
+## Scenarios
+
+Producing and consuming a message involves five stages:
+
+```
+PRODUCER
+
+Create
+  |                            CONSUMER
+  v        +--------------+                   
+Publish -> | INTERMEDIARY | -> Receive
+           +--------------+       |
+                  ^               v
+                  .            Process
+                  .               |
+                  .               v
+                  . . . . . .  Settle
+```
+
+1. The producer creates a message.
+2. The producer publishes the message to an intermediary.
+3. The consumer receives the message from an intermediary.
+4. The consumer processes the message.
+5. The consumer settles the message by notifying the intermediary that the
+   message was processed. In some cases, the message is settled before it is
+   processed, or it is settled automatically when it is received.
```

Comment:
The settlement is not necessarily related to any receive.
Some messaging scenarios involve "broadcasting" the message, so whoever is listening receives the message upon transmission, but it is possible that no consumer is registered and the message is simply discarded / settled without any consumer receiving it.

Reply:
for example: redis pub/sub, socket.io rooms, mqtt with QOS 0/1

Reply:
or alternativly, message can be transmitted to consumer, and settled without any confirmation that it was indeed received by the consumer (mqtt QOS=0).
These types of broadcast systems simply iterate on the current active subscribers, try to send the message to each of them and settle it without any feedback from the subscribers.

Reply:
I added this to the workgroup meeting notes for further discussion.

Reply:
According to the discussion in the workgroup,  I added a note that the settlement stage can be optional in exceptional cases.